builder/azure/arm/step_publish_to_shared_image_gallery.go

```diff
@@ -149,11 +148,19 @@ func (s *StepPublishToSharedImageGallery) Run(ctx context.Context, stateBag mult
 	tags := stateBag.Get(constants.ArmTags).(map[string]*string)
 
 	sharedImageGallery := getSigDestination(stateBag)
-	targetManagedImageResourceGroupName := stateBag.Get(constants.ArmManagedImageResourceGroupName).(string)
-	targetManagedImageName := stateBag.Get(constants.ArmManagedImageName).(string)
+	var sourceID string
+
+	var isManagedImage = stateBag.Get(constants.ArmIsManagedImage).(bool)
+	if isManagedImage {
+		targetManagedImageResourceGroupName := stateBag.Get(constants.ArmManagedImageResourceGroupName).(string)
+		targetManagedImageName := stateBag.Get(constants.ArmManagedImageName).(string)
 
-	managedImageSubscription := stateBag.Get(constants.ArmManagedImageSubscription).(string)
-	mdiID := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/images/%s", managedImageSubscription, targetManagedImageResourceGroupName, targetManagedImageName)
+		managedImageSubscription := stateBag.Get(constants.ArmManagedImageSubscription).(string)
+		sourceID = fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/images/%s", managedImageSubscription, targetManagedImageResourceGroupName, targetManagedImageName)
+	} else {
+		var imageParameters = stateBag.Get(constants.ArmImageParameters).(*compute.Image)
+		sourceID = *imageParameters.SourceVirtualMachine.ID
+	}
```

Comment:
ok, after reading more everything is making sense now! just one suggestion - why not always use vm as sourceId instead of using managedImage when customer has asaked for managed image also.

Reply:
>why not always use vm as sourceId instead of using managedImage when customer has asaked for managed image also.

To preserve backward compatibility

Reply:
ok, sounds good